network/p2p/p2pbuilder/gossipsub/gossipSubBuilder.go

```diff
@@ -92,14 +91,42 @@ func (g *Builder) SetGossipSubConfigFunc(gossipSubConfigFunc p2p.GossipSubAdapte
 	g.gossipSubConfigFunc = gossipSubConfigFunc
 }
 
-// SetGossipSubPeerScoring sets the gossipsub peer scoring of the builder.
-// If the gossipsub peer scoring flag has already been set, a fatal error is logged.
-func (g *Builder) SetGossipSubPeerScoring(gossipSubPeerScoring bool) {
-	if g.gossipSubPeerScoring {
-		g.logger.Fatal().Msg("gossipsub peer scoring has already been set")
+// EnableGossipSubScoringWithOverride enables peer scoring for the GossipSub pubsub system with the given override.
+// Any existing peer scoring config attribute that is set in the override will override the default peer scoring config.
+// Anything that is left to nil or zero value in the override will be ignored and the default value will be used.
+// Note: it is not recommended to override the default peer scoring config in production unless you know what you are doing.
+// Production Tip: use PeerScoringConfigNoOverride as the argument to this function to enable peer scoring without any override.
+// Args:
+// - PeerScoringConfigOverride: override for the peer scoring config- Recommended to use PeerScoringConfigNoOverride for production.
+// Returns:
+// none
+func (g *Builder) EnableGossipSubScoringWithOverride(override *p2p.PeerScoringConfigOverride) {
+	g.gossipSubPeerScoring = true // TODO: we should enable peer scoring by default.
```

Comment:
Is the `TODO` still valid?

Reply:
Yes, this `TODO` means that _eventually_, peer scoring must be a default setting for the `gossipsub` rather than setting it through builder, i.e., every `gossipsub` instance we make at our codebase must have the peer scoring enabled, and disabling the peer scoring must not be an option, as it poses a security vulnerability.